protocol/x/clob/keeper/final_settlement.go

```diff
@@ -0,0 +1,73 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	indexerevents "github.com/dydxprotocol/v4-chain/protocol/indexer/events"
+	"github.com/dydxprotocol/v4-chain/protocol/indexer/indexer_manager"
+	indexershared "github.com/dydxprotocol/v4-chain/protocol/indexer/shared"
+	"github.com/dydxprotocol/v4-chain/protocol/x/clob/types"
+)
+
+// mustTransitionToFinalSettlement holds logic executed when a market transitions to FINAL_SETTLEMENT status.
+// This function will forcefully cancel all stateful open orders for the clob pair.
+func (k Keeper) mustTransitionToFinalSettlement(ctx sdk.Context, clobPairId types.ClobPairId) {
+	// Forcefully cancel all stateful orders from state for this clob pair.
+	k.mustCancelStatefulOrdersForFinalSettlement(ctx, clobPairId)
+
+	// Delete untriggered conditional orders for this clob pair from memory.
+	delete(k.UntriggeredConditionalOrders, clobPairId)
+}
+
+// mustCancelStatefulOrdersForFinalSettlement forcefully cancels all stateful orders
+// for the provided ClobPair. These orders will be removed from the memclob in PrepareCheckState.
+func (k Keeper) mustCancelStatefulOrdersForFinalSettlement(ctx sdk.Context, clobPairId types.ClobPairId) {
+	statefulOrders := k.GetAllStatefulOrders(ctx)
+	processProposerMatchesEvents := k.GetProcessProposerMatchesEvents(ctx)
+
+	// This logic is executed in EndBlocker and should not panic. This would be unexpected,
+	// but if it happens we would rather recover and continue if an order fails to be removed from state
+	// rather than halt the chain.
+	safelyRemoveStatefulOrder := func(ctx sdk.Context, orderId types.OrderId) {
```

Comment:
suggestion: `safelyRemoveStatefulOrder` is a bit of a misnomer, maybe you could instead call it `removeStatefulOrderWithoutPanicing` (or something like that)?